crates/module-system/hyperlane/tests/integration/with_agent/helpers/evm.rs

```diff
@@ -0,0 +1,403 @@
+
+pub const ANVIL_PORT: u16 = 8545;
+const TAG: &str = "v1.1.0";
+
+pub struct AnvilRunner {
+    container: ContainerAsync<AnvilNode>,
+    req_id: u64,
+    host_port: u16,
+}
+
+impl AnvilRunner {
+    pub async fn new() -> Self {
+        tracing::info!("Starting anvil container...");
+        // Hard code tag, so we don't accidental breakages
+        let container = AnvilNode::default()
+            .with_tag(TAG)
+            .with_cmd(["--port", &ANVIL_PORT.to_string()])
+            .start()
+            .await
+            .expect("failed to start anvil");
+
+        let host_port = container
+            .get_host_port_ipv4(ANVIL_PORT)
+            .await
+            .expect("Failed to get anvil port");
+        tracing::info!(container_id = ?container.id(), %host_port, "Anvil container started successfully");
+
+        Self {
+            container,
+            req_id: 0,
+            host_port,
+        }
+    }
+
+    pub fn port(&self) -> u16 {
+        self.host_port
+    }
+
+    // Cast call is used to modify state
+    pub async fn cast_call(
+        &self,
+        contract: EthAddress,
+        abi: &str,
+        args: impl AsRef<[&str]>,
+        value: Amount,
+    ) -> Vec<EvmLog> {
+        let contract = contract.to_string();
+        let value = value.to_string();
+        let command = [
+            &["cast", "send", contract.as_str(), abi][..],
+            args.as_ref(),
+            &[
+                "--value",
+                value.as_str(),
+                "--private-key",
+                RELAYER_ACCOUNT.1,
+                "--json",
+            ][..],
+        ]
+        .concat();
+
+        tracing::info!(?command, container_id = ?self.container.id(), "executing cast call");
+
+        let mut result = self
+            .container
+            .exec(ExecCommand::new(command.clone()))
+            .await
+            .unwrap();
+
+        let mut exit_code = result.exit_code().await.expect("Failed to get exit code");
+        for _ in 0..300 {
+            exit_code = result.exit_code().await.expect("Failed to get exit code");
+            if exit_code.is_some() {
+                break;
+            }
+            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+        }
+
+        tracing::info!(?command, ?exit_code, "executed cast call");
+        let output = result.stdout_to_vec().await.unwrap();
+        if exit_code != Some(0) {
+            let std_err = result.stderr_to_vec().await.unwrap();
+            panic!(
+                "Failed to cast call.\nstdout:\n{}\nstderr:\n{}",
+                String::from_utf8_lossy(&output),
+                String::from_utf8_lossy(&std_err),
+            );
+        }
+
+        let output: CallOutput = serde_json::from_slice(&output).unwrap();
+
+        output.logs
+    }
+
+    // RPC is used to query data.
+    pub async fn rpc<T: DeserializeOwned>(&mut self, method: &str, params: Value) -> T {
```

Comment:
You could use `rpc: use ethers_middleware::SignerMiddleware;`
See `fn eth_get_balance` in the evm module.

Reply:
Offloaded into follow up: https://github.com/Sovereign-Labs/sovereign-sdk/issues/1502
As PR got quite big